File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/YGValue.java
package io.github.orioncraftmc.meditate.internal;

import io.github.orioncraftmc.meditate.internal.enums.YGUnit;

public class YGValue //Type originates from: YGValue.h
{
    public float value;
    public YGUnit unit;

    public YGValue(YGUnit unit) {
        this.unit = unit;
    }

    public YGValue(float value) {
        this.value = value;
    }

    public YGValue(float value, YGUnit unit) {
        this.value = value;
        this.unit = unit;
    }

    public float getValue() {
        return value;
    }

    public YGUnit getUnit() {
        return unit;
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/internal/YGValue.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga.internal

import app.cash.redwood.yoga.internal.enums.YGUnit

internal data class YGValue(
  val value: Float,
  val unit: YGUnit,
)
